packages/office-ui-fabric-react/src/components/MessageBar/MessageBar.base.tsx

```diff
@@ -122,9 +126,9 @@ export class MessageBarBase extends BaseComponent<IMessageBarProps, IMessageBarS
   }
 
   private _renderSingleLine(): React.ReactElement<React.HTMLAttributes<HTMLAreaElement>> {
-    const { theme } = this.props;
+    const theme = this._globalTheme;
     return (
-      <div style={{ background: theme!.semanticColors.bodyBackground }}>
+      <div style={{ background: theme.semanticColors.bodyBackground }}>
```

Comment:
what if message bar is inside of an area of the page styled via Customizer/ThemeProvider. You always want message bar to use the contextual bodyBackground.

Reply:
or what if the global theme has a blue background and you still get the "before" picture

Reply:
No, the message bar should always use the global theme's bodyBackground so it can have a predictable look anywhere on the page, which is important because the colors here convey meaning.
If the global theme has a blue background or other weird colors, then the theme author needs to customize the message bar background colors so that they work on them. Same with red error text, if the background is some shade of red that makes the error text illegible, then the theme author needs to fix the error text color.
(Note that the theme author always has the ability to make message bar background colors solid instead of partially transparent.)

Reply:
What is the point of having the color keyed off of the theme at all then? Why aren't we just hard coding it to white?

Reply:
Because it's partially transparent. This is because we share slots with list highlighting, where you can highlight certain cells or rows. The partially transparent color blends it in with the page environment better. Thus, theoretically, we are already doing our due diligence in making sure these message bar colors look nice on the global bodyBackground. (We already do insure it looks fine on our oob dark themes.)